components/Search/index.js

```diff
@@ -0,0 +1,90 @@
+import algoliasearch from "algoliasearch/lite";
+import { useState, useEffect } from "react";
+import {
+  InstantSearch,
+  // SearchBox,
+  // Hits,
+  connectSearchBox,
+  connectHits,
+} from "react-instantsearch-dom";
+import Topics from "@components/Topics";
+import Link from "next/link";
+import ReactMarkdown from "react-markdown";
+import ReactMarkdownRenderers from "@utils/ReactMarkdownRenderers";
+import PublishedDateAndReadingTime from "@components/Post/PublishedDateAndReadingTime";
+
+const searchClient = algoliasearch(
+  process.env.NEXT_PUBLIC_ALGOLIA_APP_ID,
+  process.env.NEXT_PUBLIC_ALGOLIA_SEARCH_API_KEY,
+);
+
+const Hits = ({ currentRefinement, hits }) => {
+  // this is always undefined
```

Comment:
@LukyVj I've added both custom components now... this is always undefined here! I've looked through the docs and... nothing D:

Reply:
indexContextValue is empty because the component isn't in an Index widget, unrelated to currentRefinement being set. 

However, you're getting the current refinement inside a hits component, and a hit doesn't have any refinement, and also no current refinement thusly. 

Which refinement are you looking for exactly? if it's the query string, you can change connectHits to connectAutocomplete

Reply:
Hello and thank you for stopping by! 

I'm not sure the 'currentRefinement' is what I'm looking for actually. I'm looking for the value of the SearchBox inside the Hits component.

Reply:
yes, the currentRefinement of connectAutocomplete is the same as the value of the query. Alternatively you can also use connectStateResults to get the current query inside hits somewhere :)